src/main.rs

```diff
@@ -27,35 +28,36 @@ run()
 
 fn main() {
     // init device
-    let device_handle = devices::tactrix_openport::new();
+    let device_handle = Arc::new(devices::tactrix_openport::new());
     // init server
     let server = std::net::TcpListener::bind("127.0.0.1:9001").unwrap();
     // listen for connections
-    let websockets = Arc::new(Mutex::new(vec![]));
+    let mut websockets = vec![];
```

Comment:
The websockets array is owned by the main thread. the spawned threads just borrow it using `iter()` when they want to get at its contents